lib/phx_tools_web/live/phx_tools_live/index.ex

```diff
@@ -2,31 +2,60 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
   use PhxToolsWeb, :live_view
 
   alias PhxToolsWeb.Endpoint
-  alias PhxToolsWeb.InstructionComponents
-  alias PhxToolsWeb.PhxToolsLive.LandingComponent
+  alias PhxToolsWeb.PhxToolsComponents
 
   @impl Phoenix.LiveView
   def mount(_params, session, socket) do
     {:ok,
      socket
      |> assign(:seo_attributes, %{})
-     |> assign_os(session)}
+     |> assign_os_and_source_code_url(session)}
   end
 
-  defp assign_os(socket, %{"operating_system" => operating_system}),
-    do: assign(socket, :operating_system, operating_system)
+  defp assign_os_and_source_code_url(socket, %{"operating_system" => "Linux"}) do
+    socket
+    |> assign(:operating_system, "Linux")
+    |> assign(:source_code_url, source_code_url("linux"))
+  end
+
+  defp assign_os_and_source_code_url(socket, %{"operating_system" => operating_system}) do
+    socket
+    |> assign(:operating_system, operating_system)
+    |> assign(:source_code_url, source_code_url(operating_system))
+  end
 
   @impl Phoenix.LiveView
-  def handle_params(_params, _uri, socket),
-    do: {:noreply, apply_action(socket, socket.assigns.live_action)}
+  def handle_params(_params, _uri, socket) do
+    apply_action(socket, socket.assigns.live_action)
+  end
 
-  defp apply_action(socket, :linux) do
-    assign(socket, seo_attributes: %{url: url(~p"/linux")})
+  defp apply_action(socket, :index) do
+    case socket.assigns.operating_system do
+      "Linux" -> {:noreply, assign(socket, :live_action, :linux)}
+      "macOS" -> {:noreply, assign(socket, :live_action, :macOS)}
```

Comment:
```suggestion
      "macOS" -> {:noreply, assign(socket, :live_action, :macos)}
```
Are there any other places where you need to make this change?

Reply:
Checking and updating